src/reducers/vis-state-updaters.js

```diff
@@ -634,44 +634,125 @@ export const updateAnimationSpeedUpdater = (state, action) => ({
 });
 
 /**
- * Update animation current time
+ * Update animation config current time
  * @memberof visStateUpdaters
  * @param {Object} state `visState`
  * @param {Object} action action
- * @param {Number} action.value current time value of action
+ * @param {Number} action.speed current speed of animation
  * @returns {Object} nextState
  * @public
  *
  */
 
-export const playAnimationUpdater = (state, {value}) => ({
+export const playAnimationUpdater = (state, {speed}) => ({
   ...state,
   animationConfig: {
     ...state.animationConfig,
-    currentTime: value
+    currentTime: state.animationConfig.currentTime + speed
   }
 });
 
 /**
- * Enable animation domain with the min and max of timestamps from geojson
+ * Reset animation config current time to a specified value
  * @memberof visStateUpdaters
  * @param {Object} state `visState`
  * @param {Object} action action
+ * @param {Number} action.value the value current time will be set to
  * @returns {Object} nextState
  * @public
  *
  */
 
-export const enableLayerAnimationUpdater = (state, action) => {
-  const {oldLayer} = action;
-  const [minTs, maxTs] = getTimeAnimationDomain(oldLayer, state.datasets);
+export const resetAnimationUpdater = (state, {value}) => ({
+  ...state,
+  animationConfig: {
+    ...state.animationConfig,
+    currentTime: value
+  }
+});
+
+function updateAnimationDomain(state) {
+  // merge all animatable layer domain calculate the union
+  // state.layers.config.animation.domain
+  // take state.layer
+  const animatableLayers = state.layers.filter(l => l.config.animation.enabled);
+  // const domains = state.layers.map(l => l.config.animation.domain);
+
+  const minRange = [];
```

Comment:
the whole thing can be implemented with 1 reduce function:
```
const mergedDomain = animatableLayers.reduce((accu, layer) => (
    [
      Math.min(accu[0], layer.config.animation.domain[0],
      Math.max(accu[1], layer.config.animation.domain[1]
    ]
  ), [+Infinity, -Infinity]);
```

Reply:
thanks for reviewing. modified accordingly.